bootstrap_tarball.sh

```diff
@@ -1,168 +1,203 @@
 #!/bin/sh -e
 
-mkdir -p build/tarball build/stage1/bin build/stage1/lib/autogen build/prefix
 PREFIX="$(pwd)/build/stage1"
 FINALPREFIX="${FINALPREFIX:-$(pwd)/build/prefix}"
 GUILE_VERSION="${GUILE_VERSION:-3.0}"
 
-## Get the tarball, remove generated files, and patch required files.
-
-if ! [ -f build/autogen-5.18.16.tar.xz ]; then
-	echo "=== Downloading tarball ==="
-	curl https://mirrors.kernel.org/gnu/autogen/rel5.18.16/autogen-5.18.16.tar.xz -o build/autogen-5.18.16.tar.xz
-fi
-
-if ! [ -d build/src ]; then
-	echo "=== Retrieving missing files from repository ==="
-	cd build
-	curl https://git.savannah.gnu.org/cgit/autogen.git/snapshot/autogen-5.18.16.tar.gz -o autogen-5.18.16.tar.gz
-	tar -xf autogen-5.18.16.tar.gz
-	mv autogen-5.18.16 src
-	cd src
-	rm -f add-on/char-mapper/cm.tar
-	cd ../..
+main() {
+	download_release_tarball
+	download_snapshot_tarball
+	prepare_tarball
+	bootstrap_columns
+	bootstrap_getdefs
+	bootstrap_autogen
+	bootstrap_tpl_config
+	rebuild_autogen
+}
+
+download_release_tarball() {
+	## Get the tarball
+	if ! [ -f build/autogen-5.18.16.tar.xz ]; then
+		echo "=== Downloading tarball ==="
+		rm -rf build
+		mkdir -p build
+		curl https://mirrors.kernel.org/gnu/autogen/rel5.18.16/autogen-5.18.16.tar.xz -o build/autogen-5.18.16.tar.xz
+	fi
+}
+
+download_snapshot_tarball() {
+	## Grab files missing from release tarball
+	if ! [ -d build/src ]; then
+		echo "=== Retrieving missing files from repository ==="
+		cd build
+		curl https://git.savannah.gnu.org/cgit/autogen.git/snapshot/autogen-5.18.16.tar.gz -o autogen-5.18.16.tar.gz
+		tar -xf autogen-5.18.16.tar.gz
+		mv autogen-5.18.16 src
+		cd src
+		rm -f add-on/char-mapper/cm.tar
+		cd ../..
+	fi
+}
+
+prepare_tarball() {
+	## remove generated files, and patch required files.
+	if ! [ -f build/autogen-5.18.16/ZZJUNK ]; then
+		echo "=== Preparing tarball ==="
+		tar -C build -xJf build/autogen-5.18.16.tar.xz
```

Comment:
@schierlm I was trying to run in live-bootstrap. In rootless mode there is some issue because we need to run `tar` with `--no-same-owner`. So my options are:
* Move this line to the function above where download happens.
* Or add `--no-same-owner` to `tar` command. Any preferences?

Reply:
Feel free to add --no-same-owner, I assume when it works in live-bootstrap, it must be part of really old tar :)

Reply:
Actually no, we have a very recent `tar` there.

That said, it seems that GNU tar added  this option in 1999.